source/blog/2025-07-23-orm-3.4-released.md

```diff
@@ -0,0 +1,137 @@
+---
+title: "ORM 3.4.0 released with Native Lazy Objects and Property hooks support"
+authorName: Benjamin Eberlei
+authorEmail: kontakt@beberlei.de
+permalink: /2025/10/23/orm-3.4.0-released.html
+---
+
+Last week we have [released Doctrine 3.4.0](https://github.com/doctrine/orm/releases/tag/3.4.0) which primarily adds support for PHP 8.4 functionality:
+
+- [Property Hooks](https://www.php.net/manual/en/language.oop5.property-hooks.php)
+- [Native Lazy Objects](https://www.php.net/manual/en/language.oop5.lazy-objects.php)
+
+From my perspective, this marks a new era for Doctrine ORM, one that can be
+compared with the invention of Reflection based mapping in Doctrine 2.0 way
+back in 2010.
+
+For one, this now allows to use public properties for entities by default
+without having to fear consequences down the line. Expressed with our Doctrine tutorial
+example it looks like this:
+
+```
+<?php
+
+#[ORM\Entity]
+#[ORM\Table(name: 'bugs')]
+class Bug
+{
+    #[ORM\Id]
+    #[ORM\Column(type: 'integer')]
```

Comment:
Maybe the types could be removed thanks to https://www.doctrine-project.org/projects/doctrine-orm/en/3.4/reference/basic-mapping.html#php-types-mapping ?